a11y-dialog.js

```diff
@@ -1,148 +1,308 @@
+    // stays trapped inside the dialog while open, and start listening for some
+    // specific key presses (TAB and ESC)
+    document.body.addEventListener('focus', this._maintainFocus, true);
+    document.addEventListener('keydown', this._bindKeypress);
+
+    // Execute all callbacks registered for the `show` event
+    this._listeners.show.forEach(function (listener) {
+      listener(this.node, event ? event.target : void 0);
+    }.bind(this));
+  };
+
+  // Hide the dialog element, enable all the targets (siblings), restore the
+  // focus to the previously active element, stop listening for some specific
+  // key presses and fire all registered callbacks for `hide` event
+  // @param {Event} event
+  A11yDialog.prototype.hide = function (event) {
+    // If the dialog is already closed, abort
+    if (!this.shown) {
+      return;
     }
 
-    function hide () {
-      if (!that.shown) return;
+    // Set the `shown` attribute to `false`
+    this.shown = false;
 
-      that.shown = false;
-      node.setAttribute('aria-hidden', 'true');
-      main.removeAttribute('aria-hidden');
-      focusedBeforeDialog && focusedBeforeDialog.focus();
-      document.body.removeEventListener('focus', maintainFocus, true);
-      document.removeEventListener('keydown', bindKeypress);
-      dispatchEvent(node, 'dialog:hide', this);
+    // Set the `aria-hidden` attribute from the dialog element to `true`
+    this.node.setAttribute('aria-hidden', 'true');
+
+    // Iterate over the targets to enable them by remove their `aria-hidden`
+    // attribute or resetting them to their initial value
+    this._targets.forEach(function (target) {
+      var original = target.getAttribute('data-a11y-dialog-original');
+
+      if (original) {
+        target.setAttribute('aria-hidden', original);
+        target.removeAttribute('data-a11y-dialog-original');
+      } else {
+        target.removeAttribute('aria-hidden');
+      }
+    });
+
+    // If their was a focused element before the dialog was opened, restore the
+    // focus back to it
+    if (focusedBeforeDialog) {
+      focusedBeforeDialog.focus();
     }
 
-    function destroy () {
-      hide();
+    // Remove the focus event listener to the body element and stop listening
+    // for specific key presses
+    document.body.removeEventListener('focus', this._maintainFocus, true);
+    document.removeEventListener('keydown', this._bindKeypress);
+
+    // Execute all callbacks registered for the `hide` event
+    this._listeners.hide.forEach(function (listener) {
+      listener(this.node, event ? event.target : void 0);
+    }.bind(this));
+  };
+
+  // Destroy the current instance (after making sure the dialog has been hidden)
+  // and remove all associated listeners from dialog openers and closers
+  A11yDialog.prototype.destroy = function () {
+    // Hide the dialog to avoid destroying an open instance
+    this.hide();
```

Comment:
I don't understand. So the `destroy()` method doesn't destroy anything but just 'unbind' all events handlers?
Maybe should call it `off()` then.

Edit: hum, you already have an `off()` method. Maybe merge both functions and check the passed `type` attribute.